engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,386 @@
+
+            case Long:
+                final WritableLongChunk<? extends Values> longChunk = valueChunk.asWritableLongChunk();
+                return index -> longChunk.set(index, formulaOutputSource.getLong(0));
+
+            case Short:
+                final WritableShortChunk<? extends Values> shortChunk = valueChunk.asWritableShortChunk();
+                return index -> shortChunk.set(index, formulaOutputSource.getShort(0));
+
+            default:
+                final WritableObjectChunk<Object, ? extends Values> objectChunk = valueChunk.asWritableObjectChunk();
+                return index -> {
+                    Object result = formulaOutputSource.get(0);
+                    if (result instanceof RingBufferVectorWrapper) {
+                        // Handle the rare (and probably not useful) case where the formula is an identity. We need to
+                        // copy the data in the RingBuffer and store that as a DirectVector. If not, we will point to
+                        // the
+                        // live data in the ring.
+                        result = ((Vector<?>) result).getDirect();
+                    }
+                    objectChunk.set(index, result);
+                };
+        }
+    }
+
+    @Override
+    public void startTrackingPrev() {
+        outputSource.startTrackingPrevValues();
+        if (rowRedirection != null) {
+            assert maybeInnerSource != null;
+            maybeInnerSource.startTrackingPrevValues();
+        }
+    }
+
+    @Override
+    public UpdateByOperator.@NotNull Context makeUpdateContext(int affectedChunkSize, int influencerChunkSize) {
+        return new Context(affectedChunkSize, influencerChunkSize);
+    }
+
+    @Override
+    public void prepareForParallelPopulation(final RowSet changedRows) {
+        if (rowRedirection != null) {
+            assert maybeInnerSource != null;
+            ((WritableSourceWithPrepareForParallelPopulation) maybeInnerSource)
+                    .prepareForParallelPopulation(changedRows);
+        } else {
+            ((WritableSourceWithPrepareForParallelPopulation) outputSource).prepareForParallelPopulation(changedRows);
+        }
+    }
+
+    @NotNull
+    @Override
+    public Map<String, ColumnSource<?>> getOutputColumns() {
+        return Collections.singletonMap(pair.leftColumn, outputSource);
+    }
+
+    // region clear-output
+    @Override
+    public void clearOutputRows(final RowSet toClear) {
+        // if we are redirected, clear the inner source
+        if (rowRedirection != null) {
+            ChunkUtils.fillWithNullValue(maybeInnerSource, toClear);
+        } else {
+            ChunkUtils.fillWithNullValue(outputSource, toClear);
+        }
+    }
+
+    @Override
+    public void applyOutputShift(@NotNull final RowSet subRowSetToShift, final long delta) {
+        ((SparseArrayColumnSource<?>) outputSource).shift(subRowSetToShift, delta);
```

Comment:
This may break if you have a reinterpret. Be sure to test shifts + reinterpret.
I *think* you're already OK... but a test might help prove it.